src/core/WeakMap.ts

```diff
@@ -0,0 +1,38 @@
+let wmUniqueIndex = 0;
+
+export default class WeakMap<K extends object, V> {
+
+    protected _id: string;
+
+    constructor() {
+        this._id = '__\0ec_wm_' + wmUniqueIndex++ + '_' + Math.random().toFixed(5);
+    }
+
+    get(key: K): V {
+        return (this._guard(key) as any)[this._id];
+    }
+
+    set(key: K, value: V): WeakMap<K, V> {
+        (this._guard(key) as any)[this._id] = value;
```

Comment:
It's better to use
```ts
defineProperty(key, { enumerable: false, configurable: false });
```
So this property won't be iterated when doing `clone` or something else. 

Also need to check if definePropert exists